h/h_api/exceptions.py

```diff
@@ -0,0 +1,61 @@
+from h.h_api.model.json_api import JSONAPIError, JSONAPIErrorBody
+
+
+class JSONAPIException(Exception):
+    """An exception which can turn itself into a a JSON API error response."""
+
+    http_status = None
+
+    def _error_bodies(self):
+        """Get the instances of JSONAPIErrorBody representing this error."""
+        raise NotImplementedError()
+
+    def as_dict(self):
+        """Get a JSON API compatible dict representing this error."""
+        return JSONAPIError.create(self._error_bodies()).raw
```

Comment:
So I'm guessing a Pyramid exception view is going to catch these, and call `as_dict()` to get the body for the error response. And use `http_status` for the error response status code

Reply:
Yep, either that or we construct an appropriate Pyramid HTTPException in view.